Websocket segfaults when no internet connection

set up a remote, and *without running the inertia daemon*, try:

```
~/Projects/inertia-deploy-test some-branch*
❯ inertia local logs
[WARNING] Configuration version 'test' does not match your Inertia CLI version 'v0.5.0'
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1344902]

goroutine 1 [running]:
github.com/ubclaunchpad/inertia/vendor/github.com/gorilla/websocket.(*Conn).Close(0x0, 0x0, 0x1a016a0)
	/home/travis/gopath/src/github.com/ubclaunchpad/inertia/vendor/github.com/gorilla/websocket/conn.go:331 +0x22
panic(0x17f3760, 0x20ef250)
	/home/travis/.gimme/versions/go1.11.linux.amd64/src/runtime/panic.go:513 +0x1b9
github.com/ubclaunchpad/inertia/vendor/github.com/gorilla/websocket.(*Conn).NextReader(0x0, 0x0, 0xc00014baf0, 0x1028da2, 0xc0000ca490, 0xc00014bb10)
	/home/travis/gopath/src/github.com/ubclaunchpad/inertia/vendor/github.com/gorilla/websocket/conn.go:938 +0x26
github.com/ubclaunchpad/inertia/vendor/github.com/gorilla/websocket.(*Conn).ReadMessage(0x0, 0x1a016b8, 0x0, 0x0, 0x1a016a0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ubclaunchpad/inertia/vendor/github.com/gorilla/websocket/conn.go:1028 +0x2f
github.com/ubclaunchpad/inertia/cmd/host.(*HostCmd).attachLogsCmd.func1(0xc0001c2c80, 0x211da98, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ubclaunchpad/inertia/cmd/host/host.go:306 +0x256
github.com/ubclaunchpad/inertia/vendor/github.com/spf13/cobra.(*Command).execute(0xc0001c2c80, 0x211da98, 0x0, 0x0, 0xc0001c2c80, 0x211da98)
	/home/travis/gopath/src/github.com/ubclaunchpad/inertia/vendor/github.com/spf13/cobra/command.go:766 +0x2cc
github.com/ubclaunchpad/inertia/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0000d1680, 0x2e, 0xc0000a7480, 0xc0000a7488)
	/home/travis/gopath/src/github.com/ubclaunchpad/inertia/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/ubclaunchpad/inertia/vendor/github.com/spf13/cobra.(*Command).Execute(0xc0000d1680, 0x6, 0xc0000a7480)
	/home/travis/gopath/src/github.com/ubclaunchpad/inertia/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
	/home/travis/gopath/src/github.com/ubclaunchpad/inertia/main.go:14 +0x4e
```

--- a/client/client.go
+++ b/client/client.go
@@ -323,7 +323,7 @@ func (c *Client) LogsWebSocket(container string, entries int) (SocketReader, err
 		return nil, fmt.Errorf("websocket handshake failed with status %d", resp.StatusCode)
 	}
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to connect to daemon at %s: %s", url.Host, err.Error())
 	}
 	return socket, nil
 }

--- a/client/client.go
+++ b/client/client.go
@@ -322,6 +322,9 @@ func (c *Client) LogsWebSocket(container string, entries int) (SocketReader, err
 	if err == websocket.ErrBadHandshake {
 		return nil, fmt.Errorf("websocket handshake failed with status %d", resp.StatusCode)
 	}
+	if err != nil {
+		return nil, err
+	}
 	return socket, nil
 }
 

--- a/client/client_test.go
+++ b/client/client_test.go
@@ -367,6 +367,17 @@ func TestLogsWebsocket(t *testing.T) {
 	assert.Equal(t, []byte("hello world"), m)
 }
 
+func TestLogsWebsocketNoDaemon(t *testing.T) {
+	testServer := httptest.NewTLSServer(nil)
+	// close the server to test error
+	testServer.Close()
+
+	d := newMockClient(testServer)
+	_, err := d.LogsWebSocket("docker-compose", 10)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "connect: connection refused")
+}
+
 func TestUpdateEnv(t *testing.T) {
 	testServer := httptest.NewTLSServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 		rw.WriteHeader(http.StatusOK)

